[io]  Flakky tests

I have the following error from time to time. Re-executing the CI may fix the issue.

```
Execute the tests in module ./concurrency
ok  	github.com/Scalingo/go-utils/concurrency	(cached)
Execute the tests in module ./cronsetup
?   	github.com/Scalingo/go-utils/cronsetup	[no test files]
Execute the tests in module ./crypto
ok  	github.com/Scalingo/go-utils/crypto	(cached)
Execute the tests in module ./difflib
ok  	github.com/Scalingo/go-utils/difflib	(cached)
Execute the tests in module ./env
ok  	github.com/Scalingo/go-utils/env	(cached)
Execute the tests in module ./errors
ok  	github.com/Scalingo/go-utils/errors/v2	(cached)
Execute the tests in module ./etcd
?   	github.com/Scalingo/go-utils/etcd	[no test files]
Execute the tests in module ./fs
?   	github.com/Scalingo/go-utils/fs	[no test files]
Execute the tests in module ./gomock_generator
?   	github.com/Scalingo/go-utils/gomock_generator	[no test files]
?   	github.com/Scalingo/go-utils/gomock_generator/gomockgenerator	[no test files]
Execute the tests in module ./graceful
ok  	github.com/Scalingo/go-utils/graceful	4.803s
Execute the tests in module ./httpclient
ok  	github.com/Scalingo/go-utils/httpclient	(cached)
Execute the tests in module ./influx
ok  	github.com/Scalingo/go-utils/influx	(cached)
Execute the tests in module ./io
--- FAIL: TestCopier_Copy (0.96s)
--- FAIL: TestCopier_Copy/it_should_not_impact_cache_if_option_is_set (0.96s)
copier_test.go:99: 
	Error Trace:	/home/rof/src/github.com/Scalingo/go-utils/io/copier_test.go:99
	Error:      	Should be true
	Test:       	TestCopier_Copy/it_should_not_impact_cache_if_option_is_set
	Messages:   	expected diff of cache < 110MB, was 115MB
FAIL
FAIL	github.com/Scalingo/go-utils/io	1.022s
FAIL
```

--- a/io/copier.go
+++ b/io/copier.go
@@ -3,6 +3,7 @@ package io
 import (
 	"io"
 
+	"github.com/pkg/errors"
 	"golang.org/x/sys/unix"
 )
 
@@ -47,7 +48,7 @@ func NewCopier(opts ...CopierOpt) Copier {
 	return copier
 }
 
-// copyContent is highly inspired from io.Copy, but calls to fadvise have been
+// Copy is highly inspired from io.Copy, but calls to fadvise have been
 // added to prevent caching the whole content of the files during the process,
 // impacting the whole OS disk cache
 func (c Copier) Copy(dst io.Writer, src io.Reader) (int64, error) {
@@ -61,16 +62,24 @@ func (c Copier) Copy(dst io.Writer, src io.Reader) (int64, error) {
 		if nr > 0 {
 			if fdSrc, ok := src.(Fder); c.noDiskCacheRead && ok {
 				// Fadvise is a system call giving instruction to the OS about how to behave
-				// with the flag FADC_DONTNEED, it tell the OS to drop the disk cache
-				// on a given file, on a give part of the file (initial offset + end offset)
+				// with the flag FADC_DONTNEED. It tells the OS to drop the disk cache
+				// on a given file, on a given part of the file (initial offset + end offset)
 				// http://man7.org/linux/man-pages/man2/posix_fadvise.2.html
-				unix.Fadvise(int(fdSrc.Fd()), written, written+int64(nr), unix.FADV_DONTNEED)
+				errFadvise := unix.Fadvise(int(fdSrc.Fd()), written, written+int64(nr), unix.FADV_DONTNEED)
+				if errFadvise != nil {
+					err = errors.Wrap(errFadvise, "fadvise read source file")
+					break
+				}
 			}
 
 			nw, ew := dst.Write(buf[0:nr])
 			if nw > 0 {
 				if fdDst, ok := dst.(Fder); c.noDiskCacheWrite && ok {
-					unix.Fadvise(int(fdDst.Fd()), written, written+int64(nw), unix.FADV_DONTNEED)
+					errFadvise := unix.Fadvise(int(fdDst.Fd()), written, written+int64(nw), unix.FADV_DONTNEED)
+					if errFadvise != nil {
+						err = errors.Wrap(errFadvise, "fadvise write destination file")
+						break
+					}
 				}
 				written += int64(nw)
 			}
@@ -90,5 +99,6 @@ func (c Copier) Copy(dst io.Writer, src io.Reader) (int64, error) {
 			break
 		}
 	}
+
 	return written, err
 }

--- a/io/copier_test.go
+++ b/io/copier_test.go
@@ -2,6 +2,7 @@ package io
 
 import (
 	"bytes"
+	"io"
 	"os"
 	"path/filepath"
 	"testing"
@@ -39,7 +40,8 @@ func TestCopier_Copy(t *testing.T) {
 		_, err = src.WriteString("hello")
 		require.NoError(t, err)
 		// Get back the reader at the beginning of the file
-		src.Seek(0, os.SEEK_SET)
+		_, err = src.Seek(0, io.SeekStart)
+		require.NoError(t, err)
 
 		copier := NewCopier()
 		n, err := copier.Copy(dst, src)
@@ -73,7 +75,8 @@ func TestCopier_Copy(t *testing.T) {
 		}
 
 		// Get back the reader at the beginning of the file
-		src.Seek(0, os.SEEK_SET)
+		_, err = src.Seek(0, io.SeekStart)
+		require.NoError(t, err)
 
 		copier := NewCopier(WithNoDiskCache)
 		_, err = copier.Copy(dst, src)

--- a/io/copier_test.go
+++ b/io/copier_test.go
@@ -6,7 +6,6 @@ import (
 	"path/filepath"
 	"testing"
 
-	procmeminfo "github.com/guillermo/go.procmeminfo"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
@@ -76,27 +75,10 @@ func TestCopier_Copy(t *testing.T) {
 		// Get back the reader at the beginning of the file
 		src.Seek(0, os.SEEK_SET)
 
-		mi := &procmeminfo.MemInfo{}
-		mi.Update()
-		cachedBefore := (*mi)["Cached"]
-
 		copier := NewCopier(WithNoDiskCache)
 		_, err = copier.Copy(dst, src)
 		require.NoError(t, err)
 		require.NoError(t, dst.Close())
 		require.NoError(t, src.Close())
-
-		mi.Update()
-		cachedAfter := (*mi)["Cached"]
-
-		// If the cache diff is negative, we're all good
-		// If it's positive, let's consider it's noise from the system
-		// And we want to check it's not more than the 110MB of our fixture
-		if cachedAfter > cachedBefore {
-			cacheDiff := cachedAfter - cachedBefore
-			// Let's say we want to check the difference of cache is only of 110MB,
-			// indeed it can't be insured since it's global to the system
-			assert.True(t, cacheDiff < 110*1024*1024, "expected diff of cache < 110MB, was %vMB", cacheDiff/1024/1024)
-		}
 	})
 }

--- a/io/go.mod
+++ b/io/go.mod
@@ -3,6 +3,7 @@ module github.com/Scalingo/go-utils/io
 go 1.17
 
 require (
+	github.com/pkg/errors v0.9.1
 	github.com/stretchr/testify v1.8.2
 	golang.org/x/sys v0.5.0
 )

--- a/io/go.mod
+++ b/io/go.mod
@@ -3,7 +3,6 @@ module github.com/Scalingo/go-utils/io
 go 1.17
 
 require (
-	github.com/guillermo/go.procmeminfo v0.0.0-20131127224636-be4355a9fb0e
 	github.com/stretchr/testify v1.8.2
 	golang.org/x/sys v0.5.0
 )

